xds/internal/server/conn_wrapper.go

```diff
@@ -39,7 +39,7 @@ import (
 //    retrieve the configured certificate providers.
 // 3. xDS filter_chain matching logic to select appropriate security
 //    configuration for the incoming connection.
-type connWrapper struct {
+type ConnWrapper struct {
```

Comment:
I don't see a reason why this type need to be exported. If you have an exported function on this type which returns the virtual hosts, you can type assert on an anonymous interface in the server.

```golang
func (c *connWrapper) VirtualHosts() []xdsclient.VirtualHostWithInterceptors {
  return c.virtualHosts
}
```

In `routeAndProcess`, you could do:

```golang
func routeAndProcess(ctx context.Context) error {
  ...
  cw, ok := rawConn.(interface { VirtualHosts() []xdsclient.VirtualHostWithInterceptors })
  if !ok {
    // error out
  }
  vhs := cw.VirtualHosts()
  ...
}
```

Reply:
Yup, great idea! Thanks!